platform/src/main/java/org/stellar/anchor/platform/payment/observer/stellar/PaymentObservingAccountsManager.java

```diff
@@ -0,0 +1,142 @@
+    List<PaymentObservingAccount> accounts = store.list();
+    for (PaymentObservingAccount account : accounts) {
+      ObservingAccount oa =
+          new ObservingAccount(account.getAccount(), account.getLastObserved(), true);
+      upsert(oa);
+    }
+
+    // Start the eviction task
+    ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+    scheduler.scheduleAtFixedRate(
+        this::evictAndPersist, 60, getEvictPeriod().getSeconds(), TimeUnit.SECONDS);
+  }
+
+  public void evictAndPersist() {
+    Log.info("Evicting old accounts...");
+    this.evict(getEvictMaxIdleTime());
+    Log.info("Persisting accounts...");
+    for (ObservingAccount account : this.getAccounts()) {
+      store.upsert(account.account, account.lastObserved);
+    }
+  }
+
+  /**
+   * Adds an account to be observed. If the account is being observed, it will be updated.
+   *
+   * @param account The account being observed.
+   * @param expiring true if the account can be expired. false, otherwise.
+   */
+  public void upsert(String account, Boolean expiring) throws ValueValidationException {
+    if (account != null && expiring != null) {
+      upsert(new ObservingAccount(account, Instant.now(), expiring));
+    }
+  }
+
+  /**
+   * Add an account to be observed. If the account is being observed, it will be updated.
+   *
+   * @param observingAccount The account being observed.
+   */
+  public void upsert(ObservingAccount observingAccount) throws ValueValidationException {
+    if (observingAccount != null) {
+      ObservingAccount existingAccount = allAccounts.get(observingAccount.account);
+      if (existingAccount == null) {
+        allAccounts.put(observingAccount.account, observingAccount);
+        // update the database
+        store.upsert(observingAccount.account, observingAccount.lastObserved);
+      } else {
+        if (!observingAccount.canExpire.equals(existingAccount.canExpire))
+          throw new ValueValidationException(
+              String.format(
+                  "The expiring flag cannot be modified. Account=[%s]", observingAccount.account));
+        existingAccount.account = observingAccount.account;
+        existingAccount.lastObserved = observingAccount.lastObserved;
+      }
+    }
+  }
+
+  /**
+   * Gets the list of observed accounts.
+   *
+   * @return The list of observed accounts.
+   */
+  public List<ObservingAccount> getAccounts() {
+    return new ArrayList<>(allAccounts.values());
+  }
+
+  /**
+   * Look up if the account is being observed. If the account is being observed, the lastObserved
+   * timestamp of the observing account will be updated.
+   *
+   * @param account The account to be checked.
+   * @return true if the account is being observed. false, otherwise.
+   */
+  public boolean lookupAndUpdate(String account) {
+    ObservingAccount acct = allAccounts.get(account);
+    if (acct == null) return false;
+    acct.lastObserved = Instant.now();
+    return true;
+  }
```

Comment:
❗ hey, I think you should be saving the updated value of `lastObserved` in the database. 

If you don't, we could incur in problems if we're expecting multiple Sep31Transactions to go to the same account. Imagine this scenario for instance:
1. GET callbackApi/unique_address returns `Account:memo1`
2. After 6 days, GET callbackApi/unique_address returns `Account:memo2` (same account but different memo)
3. The system is re-deployed right after that.
4. The `evict` function will evict the account in one day, but it should actually hold on to it for 7 extra days.
5. If it was persisted in the database, the re-deploy would contain the updated value and would know that the account needs to be observed for 7 extra days.

Reply:
`shutdown()` function and @PreDestroy was added to flush to database. This prevents the inconsistency when the service is shutdown for re-deployment.

Since the service become stateful because of the `Map`, we should consider memory grid implementation. I created an issue to track this problem. https://github.com/stellar/java-stellar-anchor-sdk/issues/493